compiler/rustc_resolve/src/imports.rs

```diff
@@ -837,7 +837,7 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                         }
                         let key = BindingKey::new(target, ns);
                         this.update_resolution(parent, key, false, |_, resolution| {
-                            resolution.single_imports.remove(&import);
+                            resolution.single_imports.shift_remove(&import);
```

Comment:
The `unused-warnings` regression is due to removals on `single_imports`.

Reply:
I reverted it and tried to sort it: https://github.com/rust-lang/rust/pull/131213/commits/f09b6a80351551e04ba5d9dc7da9e3d7245fe725